### PR TITLE
WIP: Introduce Window base class

### PR DESCRIPTION
**Pull request by swoolcock**
On hold pending a rethink.

---

As part of the next step in #2448, this PR:

* Replaces the osuTK `IGameWindow` inheritance with a placeholder clone `ILegacyWindow`.  This will be removed when the refactor is complete.
* Introduces a new base class `Window` that `GameWindow` inherits from.
* Moves anything not specifically osuTK-related into `Window`.

Shouldn't be any breaking changes, as long as everything is still referencing via `IWindow`.

### CODE REVIEW

File: osu.Framework/Platform/BaseWindow.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+using osuTK;
+using osuTK.Input;
+using osuTK.Platform;
+using Icon = osuTK.Icon;
+
+namespace osu.Framework.Platform
+{
+    public abstract class BaseWindow : IWindow
```
Comment (peppy) [inline osu.Framework/Platform/BaseWindow.cs]: we tend to avoid the "Base" naming paradigm where possible. how about just `Window`?

**Comment by peppy** [comment]
looks good otherwise!

**Comment by swoolcock** [comment]
Updated #2448 with a draw.io design.

**Comment by swoolcock** [comment]
Closing for now while I rethink it.

**Pull request closed without merging**